src/encoded/static/components/motifs.js

```diff
@@ -74,48 +75,91 @@ export class MotifElement extends React.Component {
+        let startAlignmentNeeded = 0;
+        let endAlignmentNeeded = 0;
+        if (strand === '-') {
+            endAlignmentNeeded = +startCoordinate - +this.props.alignedStartCoordinate;
+            startAlignmentNeeded = totalLength - endAlignmentNeeded - currentLength;
+        } else {
+            startAlignmentNeeded = +startCoordinate - +this.props.alignedStartCoordinate;
+            endAlignmentNeeded = totalLength - startAlignmentNeeded - currentLength;
+        }
+
+        // Insert padding for alignment
+        const newPWM = [...PWM];
+        /* eslint no-plusplus: ["error", { "allowForLoopAfterthoughts": true }] */
+        for (let startIdx = 0; startIdx < startAlignmentNeeded; startIdx++) {
+            newPWM.unshift([0, 0, 0, 0]);
+        }
+        for (let endIdx = 0; endIdx < endAlignmentNeeded; endIdx++) {
+            newPWM.push([0, 0, 0, 0]);
+        }
+
+        // Generate the logo from the PWM object
+        this.sequenceLogos.entryPoint(this.chartdisplay, newPWM, this.d3, alignmentCoordinate, this.props.alignedStartCoordinate, this.props.alignedEndCoordinate, strand);
     }
 
     render() {
         const element = this.props.element;
         const targetList = element.targets;
-        const targetListLabel = targetList.length > 1 ? 'Targets' : 'Target';
-        const accession = element.dataset.split('/')[2];
+        const footprintList = {};
+        const pwmList = {};
+        element.datasets.forEach((d, dIndex) => {
+            const biosample = element.biosamples[dIndex];
+            const accession = element.accessions[dIndex];
+            if (element.biosamples[dIndex] !== undefined) {
+                footprintList[biosample] = d;
+            } else {
+                pwmList[accession] = d;
+            }
+        });
+
+        const targetListLabel = (targetList.indexOf(',') !== -1) ? 'Targets' : 'Target';
+        const pwmsLabel = (Object.keys(pwmList).length > 1) ? 'PWMs' : 'PWM';
+        const footprintsLabel = (Object.keys(footprintList).length > 1) ? 'Footprints' : 'Footprint';
 
         return (
-            <div className="element" id={`element${accession}`}>
+            <div className="element" id={`element${element.pwm}`}>
                 <div className={`motif-description ${this.props.shortened ? 'shortened-description' : ''}`}>
-                    {!(this.props.shortened) ?
-                        <p><a href={element.dataset}>{accession}</a></p>
-                    : null}
-                    {element.organ_slims ?
-                        <p><span className="motif-label">Organ</span>{element.organ_slims.join(', ')}</p>
-                    : null}
                     {(targetList.length > 0) ?
-                        <p><span className="motif-label">{targetListLabel}</span>{targetList.join(', ')}</p>
+                        <p><span className="motif-label">{targetListLabel}</span>{targetList}</p>
```

Comment:
You’re just concatenating the elements of `targetList` all together?

Reply:
Now that we're grouping PWMs with the same document and the same targets, we've done the concatenation already, here: https://github.com/ENCODE-DCC/regulome-encoded/pull/58/commits/343ed2200ad0a055e1bdcb58d4ec86d490f6999e#diff-45675aa6fdfa229f6bbd721839d3071bR197